nomos-services/cryptarchia-consensus/src/states.rs

```diff
@@ -47,41 +46,13 @@ impl<TxS, BxS, NetworkAdapterSettings, BlendAdapterSettings>
         let lib_block_length = lib.length();
         let lib_leader_utxos = leader.utxos().to_vec();
```

Comment:
This was accounting for any inconsistencies between the provided prunable blocks and what's in cryptarchia, e.g., if cryptarchia had in the meanwhile moved on and it would need to prune more forks. As far as this function is concerned, cryptarchia might still have stale blocks, that is why I decided that it does not hurt to call `prunable_forks` again in here. In the best case, that will return an empty iterator and the input hashset is not changed. Maybe can we add a comment that explains we are assuming cryptarchia does not have any unpruned forks?